core/GlobalState.cc

```diff
@@ -1390,10 +1388,10 @@ NameRef GlobalState::lookupNameConstant(NameRef original) const {
     if (!original.exists()) {
         return core::NameRef::noName();
     }
-    ENFORCE(original.data(*this)->kind == NameKind::UTF8 ||
-                (original.data(*this)->kind == NameKind::UNIQUE &&
-                 (original.data(*this)->unique.uniqueNameKind == UniqueNameKind::ResolverMissingClass ||
-                  original.data(*this)->unique.uniqueNameKind == UniqueNameKind::TEnum)),
+    ENFORCE(original.kind(*this) == NameKind::UTF8 ||
+                (original.kind(*this) == NameKind::UNIQUE &&
+                 (original.unique(*this)->uniqueNameKind == UniqueNameKind::ResolverMissingClass ||
+                  original.unique(*this)->uniqueNameKind == UniqueNameKind::TEnum)),
```

Comment:
These `kind ==` checks are redundant now, right, because the `.unique()` call will check that?

We might want to get rid of those for brevity.

Reply:
good point